edge-core/sonic-buildimage
# [Platform][AS9736-64D]: Fix syncd init failed during system boot-up phase for BRCM community SAI

#### Why I did it

-   syncd init failed during system boot-up phase; moreover, syslog includes the below error logs:

  
  For HwSKUs **Accton-AS9736-64D** and **Accton-AS9736-64D-100G**:

    1) Port resource multi set failed
    INFO syncd#syncd: [none] SAI_API_PORT:brcm_sai_remove_port:8683 REMOVE_PORT port_id:0x100000098, bcm_port:152, phy:257
    ERR syncd#syncd: [none] SAI_API_PORT:brcm_sai_flex_port_oper:17286 port resource multi set failed with error Invalid parameter (0xfffffffc).
    INFO syncd#syncd: [none] SAI_API_PORT:brcm_sai_remove_port:8683 REMOVE_PORT port_id:0x100000032, bcm_port:50, phy:258
    INFO syncd#supervisord: syncd 0:bcmint_ltsw_flexport_port_eligible_validate: Remap operation is not enabled on logical port 152#015
    ERR syncd#syncd: [none] SAI_API_PORT:brcm_sai_flex_port_oper:17286 port resource multi set failed with error Invalid parameter (0xfffffffc).
    ERR syncd#syncd: :- sendApiResponse: api SAI_COMMON_API_BULK_REMOVE failed in syncd mode: SAI_STATUS_FAILURE
    INFO syncd#supervisord: syncd 0:bcmint_ltsw_flexport_port_eligible_validate: Remap operation is not enabled on logical port 50#015

  For HwSKU **Accton-AS9736-64D**:

    1) Setting Rx cosq mapping failed
    CRIT syncd#syncd: [none] SAI_API_SWITCH:brcm_sai_create_switch:3295 Setting Rx cosq mapping failed with error Feature unavailable (0xfffffff0)

    2) Invalid MTU size
    ERR syncd#syncd: [none] SAI_API_ROUTER_INTERFACE:brcm_sai_create_router_interface:339 Invalid MTU size of 9100 bytes; expected range of 64 to 1514 bytes.
    ERR syncd#syncd: :- sendApiResponse: api SAI_COMMON_API_CREATE failed in syncd mode: SAI_STATUS_INVALID_PARAMETER
    ERR syncd#syncd: :- processQuadEvent: attr: SAI_ROUTER_INTERFACE_ATTR_VIRTUAL_ROUTER_ID: oid:0x3000000000044
    ERR syncd#syncd: :- processQuadEvent: attr: SAI_ROUTER_INTERFACE_ATTR_TYPE: SAI_ROUTER_INTERFACE_TYPE_LOOPBACK
    ERR syncd#syncd: :- processQuadEvent: attr: SAI_ROUTER_INTERFACE_ATTR_MTU: 9100

    3) Tunnel terminator add failed
    ERR syncd#syncd: [none] SAI_API_TUNNEL:_brcm_sai_tunnel_term_add:6375 tunnel terminator add failed with error Feature unavailable (0xfffffff0).
    ERR syncd#syncd: [none] SAI_API_NEXT_HOP:brcm_sai_create_tunnel_term_table_entry:2134 tunnel term add failed with error -2.

  For HwSKU **Accton-AS9736-64D-100G**:

    1) Disable Switch ECC Single BitBus Buffer Error Event failed
    ERR syncd#syncd: [none] SAI_API_SWITCH:brcm_sai_create_switch:3121 Disable Switch ECC Single BitBus Buffer Error Event failed with error Feature unavailable (0xfffffff0).
    ERR syncd#syncd: :- sendApiResponse: api SAI_COMMON_API_CREATE failed in syncd mode: SAI_STATUS_NOT_SUPPORTED

    2) Error configuring flex resource failed
    ERR syncd#syncd: [none] SAI_API_PORT:brcm_sai_create_port:8277 Error configuring flex resource failed with error -1.
    ERR syncd#syncd: [none] SAI_API_PORT:brcm_sai_flex_resource_config:17375 matched fec get failed with error -1.

#### How I did it

- Fine-tune BCM configuration files for Broadcom community SAI.

For HwSKUs **Accton-AS9736-64D** and **Accton-AS9736-64D-100G**:

    1) Temporarily remove 10G port settings (Ethernet512 and Ethernet513) to bypass 'port resource multi set failed'.

For HwSKU **Accton-AS9736-64D**:

    1) Setup rx_cosq_mapping_management_mode: 0 to avoid 'Rx cosq mapping failed'.
    2) Setup MAX_FRAME_SIZE: 9416 for 400G front port config to avoid 'Invalid MTU size of 9100 bytes'.
    3) Setup bcm_tunnel_term_compatible_mode: 1 to avoid 'tunnel terminator add failed'.

For HwSKU **Accton-AS9736-64D-100G**:

    1) Setup SER_ENABLE: 1 to avoid 'Switch ECC Single BitBus Buffer Error Event failed'.
    2) Setup sai_port_pmap_phy_lanes: 0 to avoid 'Error configuring flex resource failed'.

#### How to verify it

- Boot-up system with different HwSKUs **Accton-AS9736-64D** and **Accton-AS9736-64D-100G** separately, and check syncd is working normally and no see error logs.
